### DISCUSSION TITLE
Using only Permissions part of the library

### DISCUSSION THREAD

**Discussion by fbjerggaard** [JonPSmith/AuthPermissions.AspNetCore#76] [General]
Hi there!

Just stumbled upon this library in search of something to manage permissions on a project i'm working on, and it seems like a great solution to handle everything I need.

However - I already have multi-tenancy configured using [Finbuckle.MultiTenant](https://github.com/Finbuckle/Finbuckle.MultiTenant) and authentication using Identityserver.

As far as I see it, it shouldn't be a problem using Identityserver with this project, and tenants seem to be able to be disabled by setting the `TenantType` to `NotUsingTenants` - I guess that will work just fine?

Also - I am having multiple projects (microservices) that each should handle permissions - would the preferred way be to put the Permissions enum in a class library and publish it with NuGet to other projects?

**Comment by JonPSmith** [comment]
Hi @fbjerggaard,

You are quite right, you can use the AuthP for its Roles/Permissions feature. In fact the TenantType` defaults to `NotUsingTenants`.

The main issue is how you handle users. To use Roles/Permissions you need:
1. Create some Permissions & Roles
3. Create a user in the AuthP (see [this article](https://www.thereformedprogrammer.net/three-ways-to-securely-add-new-users-to-an-application-using-the-authp-library/) and in the [AuthP documentation](https://github.com/JonPSmith/AuthPermissions.AspNetCore/wiki))
3. Link your IdentityServer to the AuthP users. I haven't built a IdentityServer-to-AuthP adapter, but have a look at the AuthP's [Add New User adapter](https://github.com/JonPSmith/AuthPermissions.AspNetCore/wiki/Add-New-User-adapter). 
4. Add Roles to a user.

On your question of microservices and Permission, yes I would suggest you create a NuGet holding your permissions.

Good luck with your project.
